src/through.ml

```diff
@@ -0,0 +1,36 @@
+module Make (C : S.Cache) (DB : S.DB with type key = C.key) = struct
+  type 'a t = 'a C.t
+  type 'a d = 'a DB.d
+  type key = C.key
+
+  let v c = C.v c
+  let stats t = C.stats t
+  let is_empty t = C.is_empty t
+  let capacity t = C.capacity t
+  let size t = C.size t
+  let clear t = C.clear t
+  let mem t d k = C.mem t k || DB.mem d k
+  let find t d k = try C.find t k with Not_found -> DB.find d k
+
+  let find_opt t d k =
+    try Some (C.find t k)
+    with Not_found -> (
+      try
+        let res = DB.find d k in
+        C.replace t k res;
+        Some res
+      with Not_found -> None)
+
+  let promote t d k =
+    (if not (C.mem t k) then
+     try C.replace t k (DB.find d k) with Not_found -> ());
+    C.promote t k
```

Comment:
What happens if `k` is not in the cache and not in the db?

Reply:
I think that we can't do nothing, because we don't know the value associated to the key.

Reply:
Oh, yes that crash because line 27 will not find (lru.ml need a rectification, it need to raise an error if k not find), so here, line 27 should be in the try

Reply:
I think we can remove `promote` from the `Cache` interface altogether, I don't think we can make it make sense for all the caching strategies anyway.